OpenRA.Mods.Common/Traits/Render/WithMakeAnimation.cs

```diff
@@ -15,40 +15,62 @@
 namespace OpenRA.Mods.Common.Traits.Render
 {
 	[Desc("Replaces the sprite during construction.")]
-	public class WithMakeAnimationInfo : ITraitInfo, Requires<WithSpriteBodyInfo>
+	public class WithMakeAnimationInfo : ITraitInfo, Requires<WithSpriteBodyInfo>, Requires<BuildingInfo>
```

Comment:
This conflicts with #13041 and shouldn't be necessary anyway (since you're only checking for the trait in INotifyCreated, not the constructor).